pkg/storage/ingest/reader.go

```diff
@@ -395,27 +406,36 @@ func (r *PartitionReader) fetchLastCommittedOffsetWithRetries(ctx context.Contex
 }
 
 func (r *PartitionReader) fetchLastCommittedOffset(ctx context.Context) (int64, error) {
```

Comment:
I'd suggest adding a comment to this method about possibility of returning special values.

Reply:
Definitely. Done in https://github.com/grafana/mimir/pull/7682/commits/009ff2a5d66d60ba64a6cccdc70eb0587595c03e.